test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -0,0 +1,303 @@
+#include "common/singleton/manager_impl.h"
+
+#include "extensions/clusters/aggregate/cluster.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/mocks/protobuf/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/test_common/environment.h"
+
+using testing::Eq;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+class AggregateClusterTest : public testing::Test {
+public:
+  AggregateClusterTest() : stats_(Upstream::ClusterInfoImpl::generateStats(stats_store_)) {}
+
+  Upstream::HostVector setupHostSet(int healthy_hosts, int degraded_hosts, int unhealthy_hosts) {
+    Upstream::HostVector hosts;
+    for (int i = 0; i < healthy_hosts; ++i) {
+      hosts.emplace_back(Upstream::makeTestHost(info_, "tcp://127.0.0.1:80"));
```

Comment:
its a bit funny to have the test generate a bunch of hosts with the same ip:port, but maybe it doesn't matter for this test? normally we increment the port number for each host or something to that effect

Reply:
Yeah, it doesn't matter for this test.